src/styles/blocks/calendar.scss

```
@@ -0,0 +1,55 @@
+.calendar {
+  max-width: 7 * $base-calendar-size + 6 * $base-calendar-gap;
+  display: flex;
+  flex-wrap: wrap;
+  gap: $base-calendar-gap;
+  padding: 10px;
+
+  @for $dayInMonth from 28 through 31 {
+    &--month-length--#{$dayInMonth} :nth-child(n + #{$dayInMonth + 1} ) {
+      display: none;
+    }
+  }
+
+  $days: (
+    'mon' 0,
+    'tue' 1 * $base-calendar-size + 1 * $base-calendar-gap,
+    'wed' 2 * $base-calendar-size + 2 * $base-calendar-gap,
+    'thu' 3 * $base-calendar-size + 3 * $base-calendar-gap,
+    'fri' 4 * $base-calendar-size + 4 * $base-calendar-gap,
+    'sat' 5 * $base-calendar-size + 5 * $base-calendar-gap,
+    'sun' 6 * $base-calendar-size + 6 * $base-calendar-gap,
+  );
+
+  @each $day, $margin in $days {
+    &--start-day--#{$day} :nth-child(1) {
+    margin-left: $margin;
+    }
+  }
+
+  &__day {
+    box-sizing: border-box;
+    width: $base-calendar-size;
+    height: $base-calendar-size;
+    background-color: $base-calendar-color;
+    border: 1px solid black;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    cursor: pointer;
+    transition-property: all 0.5s;
```

```suggestion
    transition: all 0.5s;
```
incorrect property for such value. now transition not works.